test/jdk/javax/swing/JTabbedPane/TestJTabbedPaneOpaqueColor.java

```diff
@@ -0,0 +1,236 @@
+
+            Test Case 3 (Test Tabs Opaque UI property):
+            Click to the check box 'tabs opaque' and be sure that
+              - the tabs are transparent (it must be green).
+              - Check the 'tabs opaque' and check that tabs are opaque,
+                (it must be red, except the selected tab which must be gray).
+
+            Check this behaviour for other LAFs and tab layout.
+            NOTE: For Nimbus LAF, tabs color are specific to nimbus style.""";
+
+    public static void main(String[] args) throws Exception {
+        PassFailJFrame.builder()
+            .title("JTabbedPane Tab and Content Area Color Test Instructions")
+            .instructions(INSTRUCTIONS)
+            .testTimeOut(10)
+            .rows(25)
+            .columns(60)
+            .testUI(TestJTabbedPaneOpaqueColor::createAndShowUI)
+            .build()
+            .awaitAndCheck();
+    }
+
+    private static JFrame createAndShowUI() {
+        int NUM_TABS = 5;
+        frame = new JFrame("Test JTabbedPane Opaque Color");
+        JTabbedPane tabPane = new JTabbedPane();
+        tabPane.setTabLayoutPolicy(JTabbedPane.SCROLL_TAB_LAYOUT);
+        tabPane.setTabPlacement(JTabbedPane.TOP);
+        PassFailJFrame.addTestWindow(frame);
+        PassFailJFrame.positionTestWindow(
+                frame, PassFailJFrame.Position.HORIZONTAL);
+        for (int i = 0; i < NUM_TABS; ++i) {
+            tabPane.addTab("Tab " + i , new JLabel("Content Area"));
+        }
+        JPanel panel = new JPanel(new BorderLayout());
+        panel.add(tabPane, BorderLayout.CENTER);
+        panel.setBackground(Color.green);
+        tabPane.setBackground(Color.red);
+
+        UIManager.LookAndFeelInfo[] laf = UIManager.getInstalledLookAndFeels();
+        JPanel lafButtonPanel = new JPanel(new GridLayout(1, 3));
+        for (int i = 0; i < laf.length; ++i) {
+            if (laf[i].getName().contains("Motif")
+                || laf[i].getName().contains("Windows")) {
+                continue;
+            }
+            JButton button = new JButton(laf[i].getName());
+            button.setText(laf[i].getName());
+            button.addActionListener(new MyAction());
+            lafButtonPanel.add(button);
+        }
+
+        JButton scrollButton = new JButton(new AbstractAction() {
+            public void actionPerformed(ActionEvent e) {
+                tabPane.setTabLayoutPolicy(JTabbedPane.SCROLL_TAB_LAYOUT);
+            }
+        });
+        scrollButton.setText("SCROLL layout");
+
+        JPanel layoutButtonPanel = new JPanel(new GridLayout(1, 2));
+        JButton wrapButton = new JButton(new AbstractAction() {
+            public void actionPerformed(ActionEvent e) {
+                tabPane.setTabLayoutPolicy(JTabbedPane.WRAP_TAB_LAYOUT);
+            }
+        });
+        wrapButton.setText("WRAP layout");
+
+        layoutButtonPanel.add(scrollButton);
+        layoutButtonPanel.add(wrapButton);
```

Comment:
Unless the test frame is resized or there are more number of tabs, the effect of `SCROLL_TAB_LAYOUT` or `WRAP_TAB_LAYOUT` isn't seen.

Reply:
Updated the number of test to have the effect of `SCROLL_TAB_LAYOUT` or `WRAP_TAB_LAYOUT`.